src/sentry/eventstream/kafka/protocol.py

```diff
@@ -98,3 +99,73 @@ def get_task_kwargs_for_message(value):
         )
 
     return handler(*payload[1:])
+
+
+def get_task_kwargs_for_message_from_headers(headers: Sequence[Tuple[str, Optional[bytes]]]):
+    """
+    Same as get_task_kwargs_for_message but gets the required information from
+    the kafka message headers.
+    """
+    try:
+        header_data = {k: v for k, v in headers}
+
+        def decode_str(value: Optional[bytes]) -> str:
+            assert isinstance(value, bytes)
+            return value.decode("utf-8")
+
+        def decode_optional_str(value: Optional[bytes]) -> Optional[str]:
```

Comment:
you could remove this function (and the other optional functions) by adding an `allow_none` parameter to `decode_str`, `decode_int`, `decode_bool`

Reply:
Yeah that seems a lot nicer, let me see if I can get it to cooperate with mypy

Reply:
This is pretty verbose I know but I don't know of any reasonable solution to have the return type vary depending on the value of the `allow_none` parameter. I'm going to leave it is-as so that the code that calls these functions get the proper (more restrictive) types.